src/solver/types.jl

```diff
@@ -514,87 +517,162 @@ Base.@kwdef struct EDMFXModel{
     ozone::OZ = nothing
-    """What to do with co2 for radiation (when using RRTMGP)"""
     co2::CO2 = nothing
+    insolation::IN = nothing
+    held_suarez_forcing::HS = nothing
+end
 
-    radiation_mode::RM = nothing
-    ls_adv::LA = nothing
-    external_forcing::EXTFORCING = nothing
-    scm_coriolis::EC = nothing
-    advection_test::AT = nothing
+"""
+    AtmosTurbconv
+
+Groups turbulence convection-related models and types.
+"""
+Base.@kwdef struct AtmosTurbconv{EDMFX, TCM, SAM, SEDM, SNPM, SMM, SL}
     edmfx_model::EDMFX = nothing
     turbconv_model::TCM = nothing
-    non_orographic_gravity_wave::NOGW = nothing
-    orographic_gravity_wave::OGW = nothing
-    hyperdiff::HD = nothing
-    vert_diff::VD = nothing
-    diff_mode::DM = nothing
     sgs_adv_mode::SAM = nothing
-    """sgs_entr_detr_mode == Implicit() only works if sgs_adv_mode == Implicit()"""
     sgs_entr_detr_mode::SEDM = nothing
-    """sgs_nh_pressure_mode == Implicit() only works if sgs_adv_mode == Implicit()"""
     sgs_nh_pressure_mode::SNPM = nothing
-    """sgs_mf_mode == Implicit() only works if sgs_adv_mode == Implicit() and diff_mode == Implicit()"""
     sgs_mf_mode::SMM = nothing
-    viscous_sponge::VS = nothing
     smagorinsky_lilly::SL = nothing
+end
+
+"""
+    AtmosGravityWave
+
+Groups gravity wave-related models and types.
+"""
+Base.@kwdef struct AtmosGravityWave{NOGW, OGW}
+    non_orographic_gravity_wave::NOGW = nothing
+    orographic_gravity_wave::OGW = nothing
+end
+
+"""
+    AtmosSponge
+
+Groups sponge-related models and types.
+"""
+Base.@kwdef struct AtmosSponge{VS, RS}
+    viscous_sponge::VS = nothing
     rayleigh_sponge::RS = nothing
+end
+
+"""
+    AtmosSurface
+
+Groups surface-related models and types.
+"""
+Base.@kwdef struct AtmosSurface{ST, SM, SA}
     sfc_temperature::ST = nothing
-    insolation::IN = nothing
-    """Whether to apply surface flux tendency (independent of surface conditions)"""
-    disable_surface_flux_tendency::Bool = false
     surface_model::SM = nothing
     surface_albedo::SA = nothing
-    numerics::NUM = nothing
 end
 
+# Add broadcastable for the new grouped types
+Base.broadcastable(x::SCMSetup) = tuple(x)
+Base.broadcastable(x::AtmosWater) = tuple(x)
+Base.broadcastable(x::AtmosRadiation) = tuple(x)
+Base.broadcastable(x::AtmosTurbconv) = tuple(x)
+Base.broadcastable(x::AtmosGravityWave) = tuple(x)
+Base.broadcastable(x::AtmosSponge) = tuple(x)
+Base.broadcastable(x::AtmosSurface) = tuple(x)
+
+struct AtmosModel{H, SCM, R, TC, GW, HD, VD, SP, SU, NU}
+    hydrology::H
+    scm_setup::SCM
+    radiation::R
+    turbconv::TC
+    gravity_wave::GW
+    hyperdiff::HD
+    vert_diff::VD
+    sponge::SP
+    surface::SU
+    numerics::NU
+
+    """Whether to apply surface flux tendency (independent of surface conditions)"""
+    disable_surface_flux_tendency::Bool
+end
+
+# Map grouped struct types to their names in AtmosModel struct
+const ATMOS_MODEL_GROUPS = (
+    (AtmosWater, :hydrology),
```

Comment:
any reason to name it `hydrology` rather than `water`?

Reply:
Thanks for catching that, I should have renamed it when I renamed AtmosHydrology -> AtmosWater